validate_markdown_metadata.py

```diff
@@ -53,18 +58,31 @@ def is_field_valid_type(field_name, field_value, expected_type, file_path):
             if isinstance(expected_type, tuple)
             else expected_type.__name__
         )
-        raise ValueError(f"'{field_name}' must be a {expected_types} in {file_path}")
+        errors.append(f"'{field_name}' must be a {expected_types} in {file_path}")
 
 # Validates the required fields in the frontmatter
-def validate_fields(frontmatter, file_path):
+def validate_fields(frontmatter, file_path, errors):
     # 'title': Must be a string
-    is_field_valid_type('title', frontmatter['title'], str, file_path)
+    if 'title' not in frontmatter:
```

Comment:
I thought you had code to use is_field_valid_type to append to the errors rather than have 4 cases of roughly the same if/else structure?  I may be misremembering.  In any case I think this could be far more succinct